### ISSUE TITLE
The correctness rule is overly broad

### ISSUE THREAD

**Issue by seancorfield** [clj-holmes/clj-holmes-rules#41]
The following is flagged incorrectly:
```clojure
(def foo {:bingo {:require true}})
```
as:
```
Schema require typo | Typo on schema declaration using :require instead of :required. 
```
See https://github.com/clj-holmes/clj-watson/pull/80 for where this occurred.

**Comment by samumbach** [comment]
@seancorfield @mthbernardes I think I've tracked this down to a subtle issue in shape-shifter. I'm working on a potential fix in this branch: https://github.com/clj-holmes/shape-shifter/compare/main...samumbach:shape-shifter:specified-map-keys-must-be-present

**Comment by seancorfield** [comment]
It seems like that won't work correctly if `key` is a vector/sequence of more than one element? `(get-in foo [:bar :quux])` finds a nested element's value but `(find foo :bar)` finds the top-level element. Is `key` in that context always a single-element vector/sequence?

**Comment by samumbach** [comment]
Fair point @seancorfield -- I suspect neither `find` nor `get-in` will work in the general case. In `build-hash-map-fn`, `key` is a _spec_ for the key; when the pattern is simple, that spec is a set containing a single item.

In my testing just now, I can't figure out how to get shape-shifter to match anything other than a keyword as a key in a map -- symbols, numbers, $ wildcards, true, false, nil keys all throw an exception when building the spec; aggregate (vector, set, list, map) keys throw an exception when `s/valid?` is called.

I don't think my change to shape-shifter introduces any _new_ assumptions (though it's fair to ask if the existing assumptions are desired/intentional).
